include/oneapi/dpl/internal/dynamic_selection_impl/sycl_scheduler.h

```diff
@@ -85,22 +86,16 @@ namespace experimental {
     auto submit(SelectionHandle h, Function&& f, Args&&... args) {
       using PropertyHandle = typename SelectionHandle::property_handle_t;
       auto w = new async_wait_impl_t<PropertyHandle>(h.get_property_handle(), f(h.get_native(), std::forward<Args>(args)...));
-      waiters_.push(w);
+      waiters.add_waiter(w);
       return *w;
     }
 
     auto get_wait_list(){
-       std::list<async_wait_t*> wlist;
-       waiters_.pop_all(wlist);
-       return wlist;
+      return &waiters;
```

Comment:
This doesn't seem to be destructive. The old implementation removed everything from the list, placing all of the active submission into the new wlist. This just directly returns a pointer to the container.